### PR TITLE
baseline "com.palantir.baseline:gradle-baseline-java-config:x.y.z@zip" no longer required

### LINKED ISSUE #193: make dependencies.baseline default to same version of configuration files
Currently, it is required to specify the baseline configuration files in dependencies block. Can we make it default to the same version of configuration files if it is not being set? The benefit is simplifying the config block since it is fairly common that we use the same version for the plugin and configuration file. 

```
dependencies {
    // Adds a dependency on the Baseline configuration files. Typically use 
    // the same version as the plugin itself.
    baseline "com.palantir.baseline:gradle-baseline-java-config:<version>@zip"
}
```

### PR DESCRIPTION
**Pull request by iamdanfox**
## Before this PR

Every user of baseline has the same version number in _two places_ in their build.gradle.

1. once in the buildscript dependency
2. again in the `baseline` configuration

(This functionality was originally implemented so that users could substitute their own config but this has not been heavily used, and is really just boilerplate at this point.)

## After this PR

The gradle-baseline-java-config dependency is assumed by default, so users no longer need to specify it explicitly! Note: users can still override this if they choose.

```diff
-dependencies {
-    baseline "com.palantir.baseline:gradle-baseline-java-config:x.y.z@zip"
-}
```

fixes #193, #216

### CODE REVIEW

**Comment by ferozco** [comment]
I thought we had explicitly wanted this to decouple the configuration from the plugin. What prompted the change in posture?

**Comment by iamdanfox** [comment]
Posture has not changed - the configuration remains decoupled from the plugin.  The only thing changed in this PR is that every repo that uses the default baseline dep (ie same version as the plugin) can delete three lines from their build.gradle.

File: gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineConfigIntegrationTest.groovy [hunk 1]
```diff
@@ -54,7 +54,7 @@ class BaselineConfigIntegrationTest extends AbstractPluginTest {
 
         then:
         with('baselineUpdateConfig').buildAndFail().output.contains(
-                "Expected to find exactly one config dependency in the 'baseline' configuration, found: []")
+                "Could not find com.palantir.baseline:gradle-baseline-java-config:.")
```
Comment (ferozco) [inline gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineConfigIntegrationTest.groovy]: could we add a passing test where we don't add the dependency explicitly?

**Review by ferozco — APPROVED** [review]
looks good! can we excavate to remove the cruft from users repos as a follow up?

**Comment by iamdanfox** [comment]
Yeah definitely... I think I might add one log.warn so that users realise they can remove it themselves too

**Comment by robert3005** [comment]
@dfox - I was about to make a pr for this. Can we instead use default dependencies? Or at least unify behaviour with baseline error prone? Happy to discuss and probably the ideal solution involves teaching buildscript dependencies about nebula recommendations

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineConfig.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        if (!rootProject.equals(rootProject.getRootProject())) {
+            throw new IllegalArgumentException(
+                    BaselineConfig.class.getCanonicalName() + " plugin can only be applied to the root project.");
+        }
+
+        Configuration configuration = rootProject.getConfigurations().create("baseline");
+
+        // users can still override this default dependency, it just reduces boilerplate
+        Optional<String> version = Optional.ofNullable(getClass().getPackage().getImplementationVersion());
+        configuration.defaultDependencies(d -> d.add(project.getDependencies().create(String.format(
```
Comment (iamdanfox) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineConfig.java:40]: @robert3005 this is the `defaultDependencies` approach that I'm currently using... did you have something else in mind?

**Pull request merged**